app/controllers/rso_controller.rb

```diff
@@ -32,8 +37,37 @@ def add_keyword
   end
 
   def delete_keyword
-    Keyword.find(params[:keyword_id]).destroy
-    flash[:success] = "Keyword deleted successfully"
+    if(Keyword.find(params[:keyword_id]).destroyed?)
+      flash[:success] = "Keyword deleted successfully"
+    else
+      flash[:error] = "RSO not deleted"
+    end
+
+    redirect_to action: "manage"
+  end
+
+  def add_officer
+    newOfficer = Officer.find(params[:officer_id])
+    rso = Rso.find(params[:rso_id])
+    begin
+      rso.officers << newOfficer
+      flash[:success] = "Officer added to RSO successfully"
+    rescue ActiveRecord::RecordInvalid => invalid
+      flash[:error] = "Officer not added to RSO: " + invalid.record.errors.full_messages.join(", ")
+    end
+
+
+    redirect_to action: "manage"
+  end
+
+  def delete_officer
```

Comment:
I thought "delete" removed the relationship between the two objects, whereas "destroy" removed the object itself.